sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -681,49 +682,50 @@ public static bool ActivitySourceHasListeners(object? activitySource)
                     ActivityContextType == null ||
                     ActivityKindType == null)
                 {
-                    ActivitySourceStartActivityMethod = (_, _, _, _, _, _) => null;
+                    ActivitySourceStartActivityMethod = (_, _, _, _, _, _, _) => null;
                 }
                 else
                 {
                     var method = ActivitySourceType?.GetMethod("StartActivity", BindingFlags.Instance | BindingFlags.Public, null, new[]
                     {
                         typeof(string),
                         ActivityKindType,
-                        ActivityContextType,
+                        typeof(string),
                         typeof(IEnumerable<KeyValuePair<string, object>>),
                         typeof(IEnumerable<>).MakeGenericType(ActivityLinkType),
                         typeof(DateTimeOffset)
                     }, null);
 
                     if (method == null)
                     {
-                        ActivitySourceStartActivityMethod = (_, _, _, _, _, _) => null;
+                        ActivitySourceStartActivityMethod = (_, _, _, _, _, _, _) => null;
                     }
                     else
                     {
                         var sourceParameter = Expression.Parameter(typeof(object));
                         var nameParameter = Expression.Parameter(typeof(string));
                         var kindParameter = Expression.Parameter(typeof(int));
+                        var parentIdParameter = Expression.Parameter(typeof(string));
                         var startTimeParameter = Expression.Parameter(typeof(DateTimeOffset));
                         var tagsParameter = Expression.Parameter(typeof(ICollection<KeyValuePair<string, object>>));
                         var linksParameter = Expression.Parameter(typeof(IList));
                         var methodParameter = method.GetParameters();
-                        ActivitySourceStartActivityMethod = Expression.Lambda<Func<object, string, int, ICollection<KeyValuePair<string, object>>?, IList?, DateTimeOffset, Activity?>>(
+                        ActivitySourceStartActivityMethod = Expression.Lambda<Func<object, string, int, string?, ICollection<KeyValuePair<string, object>>?, IList?, DateTimeOffset, Activity?>>(
                             Expression.Call(
                                 Expression.Convert(sourceParameter, method.DeclaringType!),
                                 method,
                                 nameParameter,
                                 Expression.Convert(kindParameter,  methodParameter[1].ParameterType),
-                                Expression.Default(ActivityContextType),
```

Comment:
We were previously using the default ActivityContextType, so the change to use the overload that does not include this parameter appears safe.